tests/testthat/test-namespace.R

```diff
@@ -394,32 +394,63 @@ test_that("can extract non-imports from namespace preserving source", {
   expect_equal(namespace_exports(path), lines[c(1:3, 5)])
 })
 
-test_that("Invalid imports throw a helpful error", {
+test_that("invalid imports generate correct declarations", {
+  # No matched functions --> no output
+  block <- "
+    #' @importFrom utils InvalidUtilsFunction
+    NULL
+  "
+  expect_message(out <- roc_proc_text(namespace_roclet(), block))
+  expect_equal(out, character())
+
+  # Matched functions --> only drop unmatched functions
   block <- "
     #' @importFrom utils head InvalidUtilsFunction
     NULL
   "
-  expect_snapshot(out <- roc_proc_text(namespace_roclet(), block))
+  expect_message(out <- roc_proc_text(namespace_roclet(), block))
   expect_equal(out, "importFrom(utils,head)")
+})
+
+test_that("invalid imports generate helpful message", {
+  block <- "
+    #' @importFrom utils head InvalidUtilsFunction1
+    NULL
+  "
+  expect_snapshot(out <- roc_proc_text(namespace_roclet(), block))
 
-  # pluralization
   block <- "
     #' @importFrom utils head InvalidUtilsFunction1 InvalidUtilsFunction2
     NULL
   "
   expect_snapshot(out <- roc_proc_text(namespace_roclet(), block))
-  expect_equal(out, "importFrom(utils,head)")
+})
 
-  # If the package is not available at roxygenize() run time, nothing we can do
+test_that("nothing we can do if package isn't installed", {
   block <- "
     #' @importFrom AnUnknownUnavailablePackage Unchecked
     NULL
   "
-  expect_snapshot(out <- roc_proc_text(namespace_roclet(), block))
+  expect_no_message(out <- roc_proc_text(namespace_roclet(), block))
   expect_equal(out, "importFrom(AnUnknownUnavailablePackage,Unchecked)")
-
 })
 
+test_that("non-syntactic imports can use multiple quoting forms", {
+  lines <- c(
+    "#' @importFrom stringr %>%",
+    "#' @importFrom stringr `%>%`",
+    "#' @importFrom stringr '%>%'",
+    "#' @importFrom stringr \"%>%\"",
+    "NULL"
+  )
+
+  import <- expect_no_warning(roc_proc_text(namespace_roclet(), lines))
+  expect_equal(import, c(
+    "importFrom(stringr,\"%>%\")",
```

Comment:
for my edification, any reason to prefer this to

```suggestion
    'importFrom(stringr,"%>%")',
```

I like this for keeping the 3 lines of code aligned vertically / no need to visually parse escapes.

Reply:
I don't have particularly strong feelings here but I like that currently all the strings use `"`, and are consistent with the values in `lines`. I saw you used `'` for the input `"%>%"`, but I didn't like having to escape `#'`.